src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -41,59 +38,60 @@ sealed class MessagePump : IPushMessages, IDisposable
         // Stop
         TaskCompletionSource<bool> connectionShutdownCompleted;
 
-        public MessagePump(ConnectionFactory connectionFactory, MessageConverter messageConverter, string consumerTag, ChannelProvider channelProvider, QueuePurger queuePurger, TimeSpan timeToWaitBeforeTriggeringCircuitBreaker, int prefetchMultiplier, ushort overriddenPrefetchCount)
+        public MessagePump(ConnectionFactory connectionFactory, IRoutingTopology routingTopology, MessageConverter messageConverter, string consumerTag,
+            ChannelProvider channelProvider, TimeSpan timeToWaitBeforeTriggeringCircuitBreaker,
+            Func<int, int> prefetchCountCalculation, ReceiveSettings settings,
+            Action<string, Exception> criticalErrorAction)
         {
             this.connectionFactory = connectionFactory;
             this.messageConverter = messageConverter;
             this.consumerTag = consumerTag;
             this.channelProvider = channelProvider;
-            this.queuePurger = queuePurger;
-            this.timeToWaitBeforeTriggeringCircuitBreaker = timeToWaitBeforeTriggeringCircuitBreaker;
-            this.prefetchMultiplier = prefetchMultiplier;
-            this.overriddenPrefetchCount = overriddenPrefetchCount;
+            this.prefetchCountCalculation = prefetchCountCalculation;
+            this.settings = settings;
+            this.criticalErrorAction = criticalErrorAction;
+
+            if (settings.UsePublishSubscribe)
+            {
+                Subscriptions = new SubscriptionManager(connectionFactory, routingTopology, settings.ReceiveAddress);
```

Comment:
If the thing that creates a message pump has access to all the things required to create a `SubscriptionManager` it seems better to me for it to create one if needed instead. That means you can stop passing in `routingTopology` just to be able to create this.

Reply:
Good point. I'll discuss that with other TF members to make this consistent across transports.